src/lib/Microsoft.Health.Events/EventCheckpointing/StorageCheckpointClient.cs

```diff
@@ -23,17 +24,25 @@ namespace Microsoft.Health.Events.EventCheckpointing
 {
     public class StorageCheckpointClient : ICheckpointClient
     {
-        private ConcurrentDictionary<string, Checkpoint> _checkpoints;
-        private ConcurrentDictionary<string, int> _lastCheckpointTracker;
-        private int _lastCheckpointMaxCount;
-        private BlobContainerClient _storageClient;
-        private ITelemetryLogger _log;
-
-        public StorageCheckpointClient(BlobContainerClient containerClient, StorageCheckpointOptions options, ITelemetryLogger log)
+        private readonly string _blobCheckpointPrefix;
+        private readonly string _blobPath;
+        private readonly ConcurrentDictionary<string, Checkpoint> _checkpoints;
+        private readonly int _lastCheckpointMaxCount;
+        private readonly ConcurrentDictionary<string, int> _lastCheckpointTracker;
+        private readonly ITelemetryLogger _log;
+        private readonly BlobContainerClient _storageClient;
+
+        public StorageCheckpointClient(BlobContainerClient containerClient, StorageCheckpointOptions options, EventHubClientOptions eventHubClientOptions, ITelemetryLogger log)
         {
             EnsureArg.IsNotNull(containerClient);
             EnsureArg.IsNotNull(options);
-            BlobPrefix = options.BlobPrefix;
+            EnsureArg.IsNotNull(eventHubClientOptions);
+
+            // Blob path for checkpoints includes the event hub name to scope the checkpoints per source event hub.
+            _blobCheckpointPrefix = $"{options.BlobPrefix}/checkpoint/";
```

Comment:
I'm not too familiar with the `default` keyword. My assumption is that `default` here would evaluate to `null` since `eventHubClientOptions` is a reference type.  If that is the case, then the check isn't needed as you already check that via EnsureArg above.

Reply:
correct, EnureArg.IsNotNull should do it.  If it was a struct then this would be different.

Reply:
Oops , my bad, meant to remove it. Thanks for catching.